books/spiders/book.py

```diff
@@ -0,0 +1,51 @@
+import scrapy
+from scrapy import Request
+from scrapy.http import Response
+
+
+class BookSpider(scrapy.Spider):
+    name = "book"
+    allowed_domains = ["books.toscrape.com"]
+    start_urls = ["https://books.toscrape.com/"]
+
+    def parse(self, response: Response, **kwargs):
+        for book in response.css(".product_pod"):
+            detail_url = response.urljoin(book.css("a::attr(href)").get())
+            yield Request(
+                url=detail_url,
+                callback=self.parse_detailed_info,
```

Comment:
The callback function is set to `self.parse_detailed_info`, but the defined method is `parse_book_details`. Ensure that the callback function name matches the method name to avoid runtime errors.